server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java

```diff
@@ -817,6 +830,30 @@ public Settings archiveUnknownOrInvalidSettings(
         }
     }
 
+    public Settings removeUnknownSettings(Settings settings) {
```

Comment:
Add javadoc?

Reply:
BTW, this method is `Version` agnostic, I decided to make it more universal in case we need version independent cleanup in the future. 

Also updated test in https://github.com/crate/crate/pull/16356/commits/5dd1269381f2f06f6841085ddf98796d02e3da06.

Should I instead make it remove**Old**UknownVersions and pass `Version`?